torch/csrc/distributed/rpc/tensorpipe_utils.cpp

```diff
@@ -96,26 +96,44 @@ std::tuple<tensorpipe::Message, TensorpipeWriteBuffers> tensorpipeSerialize(
     if (!tensorData.storageHasDeleter()) {
       std::vector<char> storageData(
           tensorData.data(), tensorData.data() + tensorData.sizeInBytes());
-      tpMessage.tensors.push_back(tensorpipe::Message::Tensor{
-          tensorpipe::CpuBuffer{storageData.data(), storageData.size()},
-          std::move(metadata)});
+      tensorpipe::CpuBuffer buffer;
+      buffer.ptr = storageData.data();
+      buffer.length = storageData.size();
+
+      tensorpipe::Message::Tensor tensor;
+      tensor.buffer = buffer;
+      tensor.metadata = std::move(metadata);
+
+      tpMessage.tensors.push_back(std::move(tensor));
```

Comment:
Would I be correct if I assume no real change here, and this PR breaks the packed line into separate lines?

Reply:
Right, this diff simply avoids relying on the order in which fields are defined in `tensorpipe::Message::Tensor` and `tensorpipe::{Cpu,Cuda}Buffer` for initialization, so that when we change the definition of those so that the `Tensor` contains the `length` rather than the `{Cpu,Cuda}Buffer`, we can update PyTorch and TensorPipe separately (before finally removing the `{Cpu,Cuda}Buffer::length` in a subsequent PR.